### PR TITLE
feat: add managed identity support to auto-validate secrets background job

### LINKED ISSUE #158: Add managed identity authentication for auto-invalidate secret store on new secret version background job
**Is your feature request related to a problem? Please describe.**
Currently, we only support connection string support when registering an Azure Service Bus topic subscription that receives 'new secret version created' events to invalidate the Arcus secret store cache.

**Describe the solution you'd like**
Add support for managed identity with the `AddCloudEventBackgroundJobUsingManagedIdentity` overload. Make sure to test this with our `TemporaryEnvironmentVariable` so that the secret values are retrieved from the environment. 
Take a look at how the `AddCloudEventBackgroundJobUsingManagedIdentity` is tested for an example.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Add managed identity support to the auto-invalidate secret background job.

Closes #158

### CODE REVIEW

File: src/Arcus.BackgroundJobs.CloudEvents/Extensions/IServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -138,7 +138,7 @@ public static class IServiceCollectionExtensions
         /// </summary>
         /// <param name="services">The services collection to add the job to.</param>
         /// <param name="serviceBusNamespace">The Service Bus namespace to connect to. This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
-        /// <param name="topicName">The name of the Azure Service Bus Topic to process.</param>
+        /// <param name="topicName">The name of the Azure Service Bus Topic to process the received event.</param>
```
Comment (fgheysels) [inline src/Arcus.BackgroundJobs.CloudEvents/Extensions/IServiceCollectionExtensions.cs:141]: I believe it is the name of the topic that contains the messages that we must proces ?
```suggestion
        /// <param name="topicName">The name of the Azure Service Bus Topic that contains the messages that must be processed.</param>
```
?

File: src/Arcus.BackgroundJobs.KeyVault/Extensions/IServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -20,6 +21,135 @@ namespace Microsoft.Extensions.DependencyInjection
...
+        /// </summary>
+        /// <remarks>
+        ///     Make sure that the application has the Arcus secret store configured correctly.
+        ///     For on the Arcus secret store: <a href="https://security.arcus-azure.net/features/secret-store" />.
+        /// </remarks>
+        /// <param name="services">The services collection to add the job to.</param>
+        /// <param name="topicName">The name of the Azure Service Bus Topic to process the received event.</param>
+        /// <param name="subscriptionNamePrefix">The name of the Azure Service Bus subscription that will be created to receive Key Vault events.</param>
+        /// <param name="serviceBusNamespace">The Service Bus namespace to connect to. This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="clientId">
```
Comment (fgheysels) [inline src/Arcus.BackgroundJobs.KeyVault/Extensions/IServiceCollectionExtensions.cs:63]: The clientId parameter actually represents a user-assigned managed identity ?  Shouldn't we then give it another name ?
If I see clientId, I always think that I need a corresponding client-secret.

Comment (stijnmoreels) [reply src/Arcus.BackgroundJobs.KeyVault/Extensions/IServiceCollectionExtensions.cs:63]: This is actually a copy-paste from the actual `<param/>` tag in the `ManagedidentityCredential`-stuff, so we used `clientId` in all managed identity-related stuff across Arcus.

Comment (fgheysels) [reply src/Arcus.BackgroundJobs.KeyVault/Extensions/IServiceCollectionExtensions.cs:63]: Looks like I've ignored this PR for a while.  Or maybe it didn't pop up in my assigned PR's to review list :?

**Review by fgheysels — APPROVED** [review]

**Comment by fgheysels** [comment]
Looks like I've neglected this PR for a while :?

**Comment by stijnmoreels** [comment]
> Looks like I've neglected this PR for a while :?

Yes, was working on this, just recently assigned you.

**Pull request merged**